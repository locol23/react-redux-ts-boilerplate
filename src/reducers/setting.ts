const SET_TITLE = 'setting/setTitle'

export const setTitle = (title: string) => ({
  type: SET_TITLE as typeof SET_TITLE,
  payload: title,
})

type Action = ReturnType<typeof setTitle>

const initialState = {
  title: '',
}

export type SettingState = typeof initialState

export const reducer = (state = initialState, action: Action) => {
  if (action.type === SET_TITLE) {
    return {
      ...state,
      title: action.payload,
    }
  }
  return state
}
