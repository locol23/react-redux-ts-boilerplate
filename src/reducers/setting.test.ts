import { reducer, setTitle } from './setting'

test('should return the initial state', () => {
  expect(
    reducer(undefined, { type: undefined as any, payload: '' })
  ).toMatchSnapshot()
})

test('should handle SET_TITLE', () => {
  expect(reducer(undefined, setTitle('title'))).toMatchSnapshot()
})
