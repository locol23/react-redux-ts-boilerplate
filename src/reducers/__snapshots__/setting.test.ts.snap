// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle SET_TITLE 1`] = `
Object {
  "title": "title",
}
`;

exports[`should return the initial state 1`] = `
Object {
  "title": "",
}
`;
